lib/speck.ex

```diff
@@ -310,12 +338,12 @@ defmodule Speck do
     {value, error}
   end
 
-  defp get_raw_value(map, key) do
-    raw_value = map[to_string(key)]
+  defp get_raw_value(map, key) when is_map_key(map, key), do: map[key]
+  defp get_raw_value(map, key), do: map[to_string(key)]
 
-    case raw_value do
-      nil -> map[key]
-      _   -> raw_value
-    end
+  defp merge_opts(global_opts, local_opts) do
+    Keyword.merge(global_opts, local_opts, fn _key, _global_value, local_value ->
+      local_value
+    end)
```

Comment:
❓ 

Where does `Keyword.merge(global_opts, local_opts)` fall short?

Reply:
You are right, for some reason I was under the impression `Keyword.merge` inserts instead of overriding keys.